ui/js/models/resultsets_store.js

```diff
@@ -1182,56 +1178,24 @@ treeherder.factory('ThResultSetStore', [
                                 return matcher;
                             }
                         );
-                    }
-                );
-                var symbolMap = _.object(paddedJobSymbolList, jobSymbolList);
-                paddedJobSymbolList.sort();
-                var outputOrderMap = {};
-                _.each(paddedJobSymbolList, function(paddedJob, index){
-                    outputOrderMap[symbolMap[paddedJob]] = index;
+                    });
+                });
+                platform.groups = _.sortBy(platform.groups, function(group) {
+                    if (!group.symbol.length)
+                        return undefined;
+                    return group.symbol;
                 });
-                return outputOrderMap;
             });
-        var jobGroupOrderPromise = ThJobGroupModel.get_list().
-            then(function(jobGroupList){
-                var jobGroupSymbolList = _.uniq(_.pluck(jobGroupList, 'symbol'));
-                jobGroupSymbolList.sort();
-                return _.object(
-                    jobGroupSymbolList,
-                    _.range(0, jobGroupSymbolList.length)
-                );
+            groupedJobs.platforms = _.sortBy(groupedJobs.platforms, function(platform){
+                var priority = thPlatformMap[platform.name];
+                if(priority) {
```

Comment:
Fixed (along with a bunch of other stuff). We should add an eslint rule for this.